### DISCUSSION TITLE
Improve performance, inspire from whisper-jax

### DISCUSSION THREAD

**Discussion by thewh1teagle** [ggml-org/whisper.cpp#1773] [Q&A]
I just saw the project `whisper-jax` which says it works 70x faster than original using `google-jax`.
But they use `Python` and it's not really usable at this moment for using in projects like [vibe](https://github.com/thewh1teagle/vibe)
But 70x faster is really attractive.
Do we have a way to improve `whisper.cpp` to get similar performance? what so special in `google-jax`?

**Comment by bobqianic** [comment]
> 70x faster

Whisper-jax segments audio into 30-second chunks and processes them in batches, which enhances the Compute-to-Memory Ratio. You can find more about this ratio here: [Compute-To-Memory Ratio](https://stackoverflow.com/questions/60249681/desired-compute-to-memory-ratio-op-b-on-gpu). In most cases, especially when using any kind of accelerator, the main limitation in decoding is the memory bandwidth. We can definitely implement this feature. It just might take a little time since I'm not a full-time developer here.
